### ISSUE TITLE
Infinite loop in System.Drawing.Graphics between IsVisible(Point) and IsVisible(int, int)

### ISSUE THREAD

**Issue by Brisingr-DE** [dotnet/winforms#12538]
### .NET version

.net9.0

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

.net8.0

### Issue description

https://github.com/dotnet/winforms/blob/e91e3bc158717736dc2c5d3ebfa8b6463ef88487/src/System.Drawing.Common/src/System/Drawing/Graphics.cs#L583-L585

### Steps to reproduce

1. Create a new WinForms-Project targetting .net9.0
2. add the following lines in the contructor of Form1
```c#
        var graphis = CreateGraphics();
        graphis.IsVisible(Point.Empty);
``` 
3. Run the application

**Comment by elachlan** [comment]
@Olina-Zhang can your team please test this?

**Comment by Olina-Zhang** [comment]
> [@Olina-Zhang](https://github.com/Olina-Zhang) can your team please test this?

It is a regression issue, regressed from .NET 9.0. Works well in .NET 8.0

**.NET 8.0 result:**
![Image](https://github.com/user-attachments/assets/74954f98-57c5-441f-a2b5-f87d88d99fc8)

**.NET 9.0/.NET 10.0 latest build result:**
![Image](https://github.com/user-attachments/assets/b631670f-8330-4976-93e0-3351cca30d0c)
System.StackOverflowException
  HResult=0x800703E9
  Message=Exception of type 'System.StackOverflowException' was thrown.

**Comment by elachlan** [comment]
@Olina-Zhang did your team want to fix it?

I figure you change one of the signatures to call the `float` or `PointF` overloads.

**Comment by kirsan31** [comment]
> I figure you change one of the signatures to call the `float` or `PointF` overloads.

I think `IsVisible((float)x, y)` will be a bit better 😌

**Comment by John-Qiao** [comment]
Verified the issues on latest .NET SDK 10.0.100-alpha.1.24573.1 + private dlls built from dotnet/winforms main branch, it has been fixed: can build and run project successfully now.

![Image](https://github.com/user-attachments/assets/36066ec8-f28d-43f3-8f65-b222ff52d8b4)

**Comment by Syareel-Sukeri** [comment]
Verified the issues on latest .NET SDK 9.0.200-preview.0.24569.43 + private dlls built from dotnet/winforms release/9.0 branch, the issue has been fixed.

![Image](https://github.com/user-attachments/assets/661d496e-caf8-4921-8f05-638f73e03b5a)

**Comment by lonitra** [comment]
Fix will be available in 9.0.1

**Comment by chaowendy** [comment]
Verified it in the 9.0.1 servicing test pass, this issue was fixed with the same results as above.
